packages/coreutils/src/settingregistry.ts

```diff
@@ -550,6 +563,16 @@ export class SettingRegistry implements ISettingRegistry {
 
     // Fetch and reload the data to guarantee server and client are in sync.
     const fetched = await this.connector.fetch(plugin);
+    if (fetched === undefined) {
+      throw [
+        {
+          dataPath: '',
+          keyword: 'id',
+          message: `Could not fetch settings for ${plugin}.`,
+          schemaPath: ''
+        } as ISchemaValidator.IError
+      ];
+    }
```

Comment:
@afshin Is this a reasonable error to throw?

Reply:
If you want to throw an error here, you need to put the `const fetched ...` line in a `try` block, because it can `throw`.